openedx/core/djangoapps/schedules/management/commands/sync_schedules_to_course_start.py

```diff
@@ -0,0 +1,54 @@
+"""
+Support "fixing" schedules that currently have start dates prior to the course start date.
+
+This can happen if a course start date is moved into the future and the corresponding
+schedules are not updated for any reason.
+"""
+from __future__ import print_function
+
+import logging
+import textwrap
+
+from django.core.management.base import BaseCommand
+from django.db import connection
+
+
+LOG = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """
+    Support "fixing" schedules that currently have start dates prior to the course start date.
+
+    This can happen if a course start date is moved into the future and the corresponding
+    schedules are not updated for any reason.
+    """
+
+    def add_arguments(self, parser):
+        parser.add_argument('--dry-run', default=False, action='store_true')
+        parser.add_argument('from_date')
+        parser.add_argument('course_id')
+
+    def handle(self, *args, **options):
+        query = textwrap.dedent(
+            """
+            UPDATE
+                schedules_schedule s
+            JOIN student_courseenrollment e ON e.id = s.enrollment_id
+            JOIN course_overviews_courseoverview c ON c.id = e.course_id
+            SET s.start = c.start
+            WHERE
+                s.start < c.start
+                AND s.start > %s
+                AND c.start < CURRENT_DATE()
```

Comment:
This means that we won't fix this problem for courses that haven't started yet? Why is that restriction in there?

Reply:
errr, yeah this is an older condition I had in place before the course_id was added to the where clause, we should remove this.